presto-orc/src/test/java/com/facebook/presto/orc/BenchmarkOrcDecimalReader.java

```diff
@@ -0,0 +1,143 @@
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.orc.OrcTester.Compression.NONE;
+import static com.facebook.presto.orc.OrcTester.Format.ORC_12;
+import static com.facebook.presto.orc.OrcTester.writeOrcColumn;
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+import static com.google.common.io.Files.createTempDir;
+import static io.airlift.testing.FileUtils.deleteRecursively;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.UUID.randomUUID;
+import static org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory.javaHiveDecimalObjectInspector;
+
+@SuppressWarnings("MethodMayBeStatic")
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Fork(3)
+@Warmup(iterations = 20, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 20, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkOrcDecimalReader
+{
+    public static final DecimalType DECIMAL_TYPE = createDecimalType(30, 10);
+
+    @Benchmark
+    public Object readDecimal(BenchmarkData data)
+            throws Throwable
+    {
+        OrcRecordReader recordReader = data.createRecordReader();
+        List<Block> blocks = new ArrayList<>();
+        while (recordReader.nextBatch() > 0) {
+            Block block = recordReader.readBlock(DECIMAL_TYPE, 0);
+            blocks.add(block);
+        }
+        return blocks;
+    }
+
+    @SuppressWarnings("FieldMayBeFinal")
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private File temporary;
+        private File dataPath;
+
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            temporary = createTempDir();
+            dataPath = new File(temporary, randomUUID().toString());
+
+            writeOrcColumn(dataPath, ORC_12, NONE, javaHiveDecimalObjectInspector, crateDecimalValues().iterator());
+        }
+
+        public OrcRecordReader createRecordReader()
+                throws IOException
+        {
+            OrcDataSource dataSource = new FileOrcDataSource(dataPath, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE));
+            MetadataReader metadataReader = new OrcMetadataReader();
+            OrcReader orcReader = new OrcReader(dataSource, metadataReader, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE));
+            return orcReader.createRecordReader(
+                    ImmutableMap.of(0, DECIMAL_TYPE),
+                    OrcPredicate.TRUE,
+                    DateTimeZone.forID("Asia/Katmandu"),
+                    new AggregatedMemoryContext());
+        }
+
+        private List<SqlDecimal> crateDecimalValues()
+        {
+            List<SqlDecimal> values = new ArrayList<>();
+            for (int i = 0; i < 500000; ++i) {
+                values.add(new SqlDecimal(BigInteger.valueOf(i), 10, 5));
```

Comment:
This pattern of alternating positive/negative values seems very specific. Is there are particular reason for benchmarking this? (If so, add a comment). If not, you may want to consider a more random data set.